site/src/pages/DeploySettingsPage/NotificationsPage/NotificationEvents.stories.tsx

```diff
@@ -0,0 +1,71 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import { spyOn, userEvent, within } from "@storybook/test";
+import { API } from "api/api";
+import { selectTemplatesByGroup } from "api/queries/notifications";
+import type { DeploymentValues } from "api/typesGenerated";
+import { MockNotificationTemplates } from "testHelpers/entities";
+import { NotificationEvents } from "./NotificationEvents";
+import { baseMeta } from "./storybookUtils";
+
+const meta: Meta<typeof NotificationEvents> = {
+	title: "pages/DeploymentSettings/NotificationsPage/NotificationEvents",
+	component: NotificationEvents,
+	args: {
+		defaultMethod: "smtp",
+		availableMethods: ["smtp", "webhook"],
+		templatesByGroup: selectTemplatesByGroup(MockNotificationTemplates),
+		deploymentValues: baseMeta.parameters.deploymentValues,
+	},
+	...baseMeta,
+};
+
+export default meta;
+
+type Story = StoryObj<typeof NotificationEvents>;
+
+export const NoEmailSmarthost: Story = {
+	args: {
+		deploymentValues: {
+			notifications: {
+				webhook: {
+					endpoint: "https://example.com",
+				},
+				email: {
+					smarthost: "",
+				},
+			},
+		} as DeploymentValues,
+	},
+};
+
+export const NoWebhookEndpoint: Story = {
+	args: {
+		deploymentValues: {
+			notifications: {
+				webhook: {
+					endpoint: "",
+				},
+				email: {
+					smarthost: "smtp.example.com",
+				},
+			},
+		} as DeploymentValues,
+	},
+};
+
+export const Toggle: Story = {
+	play: async ({ canvasElement }) => {
+		spyOn(API, "updateNotificationTemplateMethod").mockResolvedValue();
+		const user = userEvent.setup();
+		const canvas = within(canvasElement);
+		const option = await canvas.findByText("Workspace Marked as Dormant");
```

Comment:
I wonder if we should include the template ID as a data attribute and select on that? It'll be more stable than the name.

Reply:
We control it, yes, but we may rename this template, which will cause this test to fail mysteriously.
IDs will likely never be changed, so it's more resilient.

Reply:
This test won't fail if we rename the template because we're using mocked data. It's not an end-to-end test. I understand that using IDs makes the test more resilient, but this is a UI test in a controlled context, so I think we should test it as the user would. However, if you strongly prefer using IDs, we can switch to that approach and start a discussion with the front-end guide to establish it as a convention, since the community uses a different method. The goal is to test the UI as the user would.

Reply:
Right, but when you implement what we discussed in https://github.com/coder/coder/pull/14565/files#r1745353934 then it _will_ go out of sync, and this is an entirely preventable problem.
I'm not part of the front-end guild so it's up to you.